gallery/__init__.py

```diff
@@ -618,6 +617,96 @@ def move_dir(dir_id: int, auth_dict: Optional[Dict[str, Any]] = None):
 
     return "ok", 200
 
+@app.route("/api/file/pin/<int:file_id>", methods=['POST'])
+@auth.oidc_auth('default')
+@gallery_auth
+def pin_file(file_id: int, auth_dict: Optional[Dict[str, Any]] = None):
+    file_model = File.query.filter(File.id == file_id).first()
+
+    if file_model is None:
+        return "file not found", 404
+
+    assert auth_dict
+    if file_model.hidden and not (auth_dict['is_eboard'] or auth_dict['is_rtp']):
+        return "Permission denied", 403
+
+    if not (auth_dict['is_eboard']
+            or auth_dict['is_rtp']
+            or auth_dict['is_organizer']):
+        return "Permission denied", 403
+    
+    file_model.pinned = True
+
+    db.session.commit()
+
+    return "ok", 200
+
+@app.route("/api/dir/pin/<int:dir_id>", methods=['POST'])
+@auth.oidc_auth('default')
+@gallery_auth
+def pin_dir(dir_id: int, auth_dict: Optional[Dict[str, Any]] = None):
+    dir_model = Directory.query.filter(Directory.id == dir_id).first()
+
+    if dir_model is None:
+        return "directory not found", 404
+
+    assert auth_dict
+    if not (auth_dict['is_eboard']
+            or auth_dict['is_rtp']
+            or auth_dict['is_organizer']):
+        return "Permission denied", 403
+    
+    dir_model.pinned = True
+
+    db.session.commit()
+
+    return "ok", 200
+
+@app.route("/api/file/unpin/<int:file_id>", methods=['POST'])
+@auth.oidc_auth('default')
+@gallery_auth
+def unpin_file(file_id: int, auth_dict: Optional[Dict[str, Any]] = None):
+    file_model = File.query.filter(File.id == file_id).first()
+
+    if file_model is None:
+        return "file not found", 404
+
+    assert auth_dict
+    if file_model.hidden and not (auth_dict['is_eboard'] or auth_dict['is_rtp']):
+        return "Permission denied", 403
+
+
+    if not (auth_dict['is_eboard']
+            or auth_dict['is_rtp']
+            or auth_dict['is_organizer']):
+        return "Permission denied", 403
+    
+    file_model.pinned = False
+
+    db.session.commit()
+
+    return "ok", 200
+
+@app.route("/api/dir/unpin/<int:dir_id>", methods=['POST'])
+@auth.oidc_auth('default')
+@gallery_auth
+def unpin_dir(dir_id: int, auth_dict: Optional[Dict[str, Any]] = None):
+    dir_model = Directory.query.filter(Directory.id == dir_id).first()
+
+    if dir_model is None:
+        return "directory not found", 404
+
+    assert auth_dict
+    if not (auth_dict['is_eboard']
+            or auth_dict['is_rtp']
+            or auth_dict['is_organizer']):
+        return "Permission denied", 403
+    
```

Comment:
does this need a hidden check?

Reply:
Directories cannot be hidden, only files